libraries/AP_Notify/AP_Notify.cpp

```diff
@@ -199,6 +199,76 @@ void AP_Notify::add_backend_helper(NotifyDevice *backend)
 
 #define ADD_BACKEND(backend) do { add_backend_helper(backend); if (_num_devices >= CONFIG_NOTIFY_DEVICES_MAX) return;} while(0)
 
+void AP_Notify::set_nav_lights(void)
+{
+  AP_HAL::OwnPtr<AP_HAL::I2CDevice> dev1 =  std::move(hal.i2c_mgr->get_device(0, 0x51,100000, true, 20));
+
+  bool dev1sem = false;
+  bool dev2sem = false;
+  bool dev3sem = false;
+  bool dev4sem = false;
+
+  if(!dev1->get_semaphore()->take(10))
+    gcs().send_text(MAV_SEVERITY_CRITICAL, "dev1 sem take fail");
+  else{
+    gcs().send_text(MAV_SEVERITY_CRITICAL, "dev1 sem take good");
```

Comment:
Cut down the `send_text` to just send a message if it failed to get the semaphore for any of the devices, _or_ if any of the writes fail

Reply:
changed